08-read-write-plot.Rmd

```diff
@@ -708,6 +708,39 @@ writeRaster(x = single_layer, filename = "my_raster.tif",
 
 To learn more about the compression of GeoTIFF files, read Paul Ramsey's [comprehensive blog post on this topic](https://blog.cleverelephant.ca/2015/02/geotiff-compression-for-dummies.html).
 
+## Geographic metadata
+
+Geographic metadata are a cornerstone of geographic information management, used to describe datasets, data structures and services. 
+They help make datasets FAIR (Findable, Accessible, Interoperable, Reusable) and are defined by the ISO/OGC standards, in particular the ISO 19115 standard and underlying schemas. 
+These standards are widely used within spatial data infrastructures, handled through metadata catalogs.
+
+Geographic metadata can be managed with **geometa**, a package that allows writing, reading and validating geographic metadata according to the ISO/OGC standards. 
+It already supports various international standards for geographic metadata information, such as the ISO 19110 (Feature catalogue), ISO 19115-1 and 19115-2 (geographic metadata for vector and gridded/imagery datasets), ISO 19119 (geographic metadata for service), and ISO 19136 (Geographic Markup Language) providing methods to read, validate and write geographic metadata from R using the ISO/TS 19139 (XML) technical specification.
+To address the complexity and exhaustiveness the standards, and the advanced knowledge required to use them, complementary packages such as [geoflow](https://github.com/r-geoflow/geoflow) have been built to ease and automate the management of metadata.
+Geographic metadata can be created with **geometa** as follows, which creates and saves a metadata file:
+
+```{r 07-read-write-plot-m5, eval=FALSE}
+library(geometa)
+# create a metadata
+md = ISOMetadata$new()
+#... fill the metadata 'md' object
+# validate metadata
+md$validate()
+# XML representation of the ISOMetadata
+xml = md$encode()
+# save metadata
+md$save("my_metadata.xml")
+# read a metadata from an XML file
+md = readISO19139("my_metadata.xml")
+```
+
+The package comes with more [examples](https://github.com/eblondel/geometa/tree/master/inst/extdata/examples) and has been extended by packages such as **[geoflow](https://github.com/r-geoflow/geoflow)** to ease and automate the management of metadata
+
+In the field of standard geographic information management, the distinction between data and metadata is less clear.
+The Geography Markup Language (GML) standard and file format covers both data and metadata, for example. 
+The **geometa** package allows exporting GML (ISO 19136) objects from geometry objects modeled with **sf**.
+Such functionality allows use of geographic metadata (enabling the inclusion of metadata on detailed geographic and temporal extents, rather than simple bounding boxes, for example) and the provisioin of services that extend the GML standard (e.g., OGC Web Coverage Service).
```

Comment:
Is my update correct here @eblondel, is it more about providing or interacting with services?

Reply:
not really on the last sentence

Reply:
GML is often known as data format (the main OGC format), but many GML elements are metadata-oriented, and are used in service descriptions (enabling interoperability when interrogating OWS services with ows4R, with geometa as decoder for these GML elements), but also in dataset metadata, for which some metadata building blocks expect GML elements, it's the case of extents (geographic, temporal, vertical). 

For example The usual way to provide a geographic extent is to provide a bounding box, but it's a rough simplification of extents, and in some cases (e.g. vessel transects, routes network, etc) you want to provide a finer extent, e.g. with simplified linear geometries. All this can be provided in metadata, and this is where from the ISO 19115 you jump into the GML.
With **geometa**, we have a generic encoder/decoder of all elements, including GML, and converters from **sf**, which let you easily move from R **sf** to GML representations part of ISO 19115 metadata.